views/StatisticsView.vue

```diff
@@ -76,37 +46,208 @@ function populateStats(){
+            {{ averageRank }}
+          </div>
+          <p class="mt-3 text-sm text-gray-500">Avg. Rank</p>
+        </div>
+      </div>
+      <div class="flex justify-around w-full md:w-1/2">
+        <div class="flex flex-col items-center">
+          <div class="flex justify-center items-center mr-2 h-24 w-24 rounded-full border-8 text-2xl font-extrabold border-primary">
+            {{ averageGuesses }}
+          </div>
+          <p class="mt-3 text-sm text-gray-500">Avg. Guesses</p>
+        </div>
+        <div class="flex flex-col items-center">
+          <div class="flex justify-center items-center mr-2 h-24 w-24 rounded-full border-8 text-2xl font-extrabold border-primary">
+            {{ averageTime }}
+          </div>
+          <p class="mt-3 text-sm text-gray-500">Avg.Time</p>
+        </div>
+        <div class="flex flex-col items-center">
+          <div class="flex justify-center items-center h-24 w-24 rounded-full border-8 text-2xl font-extrabold border-primary">
+            {{ gamesPlayed }}
+          </div>
+          <p class="mt-3 text-sm text-gray-500">Games Played</p>
+        </div>
+      </div>
+    </div>
+    <div class="flex flex-col md:flex-row justify-center items-end w-full mt-10">
+      <div class="w-full md:w-1/2 mb-5 md:mb-0 flex flex-col items-center">
+        <Line :data="guessesData" :options="guessesOptions" />
+        <p class="mt-3 text-sm text-gray-500">{{ guessesTitle }}</p>
+      </div>
+      <div class="w-full md:w-1/2 flex flex-col items-center">
+        <Bar :data="ranksData" :options="ranksOptions" />
+        <p class="mt-3 text-sm text-gray-500">{{ ranksTitle }}</p>
+      </div>
     </div>
-    <Line
-        id="my-chart-id"
-        :options="{
-          responsive: true,
-          scales: {
-            x: { grid: {color: 'grey'} },
-            y: { grid: {color: 'grey'} }
-          }
-        }"
-        :data="{
-          labels: Array.from(Array(times.length).keys()),
-          datasets: [ { label: 'Times', data: times, backgroundColor: 'blue', borderColor: 'blue' },
-          { label: 'Ranks', data: ranks, backgroundColor: 'green', borderColor: 'green' },
-          { label: 'Guesses', data: guesses, backgroundColor: 'red', borderColor: 'red' } ]
-        }"
-    />
   </div>
   <div v-else class="flex flex-col items-center justify-center">
     <p>User not logged in</p>
   </div>
-  <div v-if="user" class="font-medium text-blue-600 dark:text-blue-500 hover:underline">
-    <UButton @click="populateStats()">Populate stats</UButton>
-  </div>
 </template>
+
+<script lang="ts">
+// Script to handle chart logic
+import { Line, Bar } from 'vue-chartjs'
+import {Chart as ChartJS, type ChartData, type ChartOptions, type Point, registerables} from 'chart.js'
+import type { TimedStat } from "~/model/UserModel";
+
+ChartJS.register(...registerables)
+
```

Comment:
In all this second script banner you can gain a lot of lines. 
Also shouldn't you have both script banners together (scrip-script-template instead of script-template-script) ?